src/quo2/components/dropdown.cljs

```diff
@@ -0,0 +1,150 @@
+                                                    :align-items :center
+                                                    :border-width 1
+                                                    :border-color (if @open?
+                                                                    border-color
+                                                                    (color-by-10 border-color))
+                                                    :background-color (if @open?
+                                                                        dd-color
+                                                                        (color-by-10 dd-color))})
+                                            disabled? (assoc :opacity 0.3))}
+                            disabled? (assoc :disabled true))
+     (when icon
+       [icons/icon icon {:no-color true
+                         :size 20
+                         :container-style {:margin-right spacing
+                                           :margin-top 1
+                                           :width icon-size
+                                           :height icon-size}}])
+     [text/text {:size font-size
+                 :weight :medium
+                 :font :font-medium
+                 :color :main
+                 :style {:margin-right spacing}} text]
+     [icons/icon (if @open?
+                   (if dark?
+                     :main-icons/pullup
+                     :main-icons/pullup-black)
+                   (if dark?
+                     :main-icons/dropdown-white
+                     :main-icons/dropdown-black))
+      {:no-color true
+       :size 20
+       :container-style {:width icon-size
+                         :margin-top 1
+                         :height icon-size}}]]))
+
+(defn items-comp
+  [{:keys [items on-select]}]
+  (let [items-count (count items)]
+    [:f>
+     (fn []
+       [rn/scroll-view {:style {:height "100%"}
+                        :horizontal false
+                        :nestedScrollEnabled true}
+        (doall
+         (map-indexed (fn [idx item]
+                        [rn/touchable-opacity
+                         {:key (str item idx)
+                          :style {:padding 4
+                                  :border-bottom-width (if (= idx (- items-count 1))
+                                                         0
+                                                         1)
+                                  :border-color (colors/theme-colors
+                                                 colors/black
+                                                 colors/white)
+                                  :text-align :center}
+                          :on-press #(on-select item)}
+                         [text/text {:style {:text-align :center}} item]]) items))])]))
+
+(defn dropdown [{:keys [items icon text default-item on-select size disabled? border-color dd-color]}]
+  [:f>
```

Comment:
Is the `:f` necessary here? Isn't it just the case of defining the `reagent/atom` outside the anonymous function, just like any other form-2 component?

Reply:
I think reanimated uses hooks .. And hooks need functional components